### PR TITLE
Refactor SIR <-> AST interdependencies

### PR DESCRIPTION
**Pull request by muellch**
## Technical Description

Gets rid of the SIR<-> AST circular dependencies resolving https://github.com/MeteoSwiss-APN/dawn/issues/319 and partially resolving https://github.com/MeteoSwiss-APN/dawn/issues/320.

The most important change is the separation of anything related to `class sir::VerticalRegion`, including `class ast::VerticalRegionDeclStmt`, into separate files `VerticalRegion.h` and `VerticalRegion.cpp`. In these files `class sir::VerticalRegion` remains defined as part of the `sri` namespace, while `ast::VerticalRegionDeclStmt` remains defined as part of the namespace `ast`.

Other noteworthy changes:
- `class sri::Value`, `class sri::Interval` and `class sri::Attr` have all been extracted into according header and `.cpp` files and moved to the namespacd `ast`.
- All `sir` and `iir` aliased of `ast` classes have been removed.
- A lot of `#include` directives have been cleaned up throughout the project (Thank you CLion).

Future discussions and TODOs:
- One part of https://github.com/MeteoSwiss-APN/dawn/issues/320 remains open:
https://github.com/MeteoSwiss-APN/dawn/blob/10d08b48b78133464485f119372b07f6812ca3f1/dawn/src/dawn/AST/ASTStmt.h#L504-L513
- Should other classes in the `sri` namespace, such as `class sri::StencilFunction` or `class sri::Field` also be extracted into separate files and moved into the `ast` namespace?

### CODE REVIEW

**Comment by muellch** [comment]
launch jenkins

**Comment by muellch** [comment]
launch jenkins

File: dawn/src/dawn/SIR/VerticalRegion.cpp [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+#include "dawn/SIR/ASTVisitor.h"
+#include "dawn/SIR/VerticalRegion.h"
+#include "dawn/Support/Casting.h"
+#include "dawn/Support/Format.h"
+#include "dawn/Support/StringUtil.h"
+#include <sstream>
+
+namespace dawn {
+
+CompareResult sir::VerticalRegion::comparison(const sir::VerticalRegion& rhs) const {
```
Comment (Stagno) [inline dawn/src/dawn/SIR/VerticalRegion.cpp:8]: I suggest to move it below where you open a `namespace sir`

File: dawn/src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp [hunk 2]
```diff
@@ -33,12 +33,12 @@
 namespace dawn {
 
 namespace {
-sir::Interval intervalToSIRInterval(iir::Interval interval) {
-  return sir::Interval(interval.lowerLevel(), interval.upperLevel(), interval.lowerOffset(),
+ast::Interval intervalToSIRInterval(iir::Interval interval) {
```
Comment (Stagno) [inline dawn/src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp:36]: `intervalToASTInterval`

File: dawn/src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp [hunk 3]
```diff
@@ -33,12 +33,12 @@
...
+  return ast::Interval(interval.lowerLevel(), interval.upperLevel(), interval.lowerOffset(),
                        interval.upperOffset());
 }
 
-iir::Interval sirIntervalToInterval(sir::Interval interval) {
+iir::Interval sirIntervalToInterval(ast::Interval interval) {
```
Comment (Stagno) [inline dawn/src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp:41]: `astIntervalToInterval`

**Review by Stagno — CHANGES_REQUESTED** [review]
PR is in good shape, good job! Very small changes requested.

**Comment by Stagno** [comment]
@muellch about the two points you mention in the description, I'd say yes, they could be done as followups to this PR if you still think AST refactorings are still worth your time.

File: dawn/src/dawn/SIR/VerticalRegion.cpp (hunk 1 above)
Comment (muellch) [reply dawn/src/dawn/SIR/VerticalRegion.cpp]: Done.

File: dawn/src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp (hunk 2 above)
Comment (muellch) [reply dawn/src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp]: Done.

File: dawn/src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp (hunk 3 above)
Comment (muellch) [reply dawn/src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp]: Done.

**Comment by muellch** [comment]
launch jenkins

**Review by Stagno — APPROVED** [review]

**Comment by Stagno** [comment]
@muellch Squash and merge at your will, as PR is approved and build is green

**Pull request merged**